openedx/adg/lms/applications/views.py

```diff
@@ -263,3 +266,100 @@ def get_context_data(self, request):
+        """
+        user_application_hub, _ = ApplicationHub.objects.get_or_create(user=self.request.user)
+
+        return not user_application_hub.is_written_application_completed
+
+    def handle_no_permission(self):
+        """
+        Redirects to application hub on get request or returns http 400 on post request.
+        Returns:
+            HttpResponse object.
+        """
+        if self.request.method == 'POST':
+            return HttpResponse(status=400)
+        else:
+            return redirect('application_hub')
+
+    def get(self, request):
+        """
+        Send the context data for example a list of business lines and saved user_application.
+        Returns:
+            HttpResponse object.
+        """
+        business_lines = BusinessLine.objects.all()
+        file_name = None
+
+        try:
+            registration_business_line = request.user.extended_profile.company
+        except ExtendedUserProfile.DoesNotExist:
+            registration_business_line = None
+
+        try:
+            user_application = request.user.application
+            if user_application.cover_letter_file:
+                file_name = Path(user_application.cover_letter_file.name).name
+        except UserApplication.DoesNotExist:
+            user_application = None
+
+        context = {
+            'business_lines': business_lines,
+            'user_application': user_application,
+            'registration_business_line': registration_business_line,
+            'csrf_token': get_token(request),
+            'filename': file_name
+        }
+
+        return render(request, self.template_name, context)
+
+    def post(self, request):
+        """
+        Submit user application and redirect to application hub or experience depending upon button click.
+        Returns:
+            HttpResponse object.
+        """
+        user_application, _ = UserApplication.objects.get_or_create(user=request.user)
+
+        if 'business_line' in request.POST and request.POST['business_line']:
+            business_line = BusinessLine.objects.get(id=request.POST['business_line'])
+            user_application.business_line = business_line
+
+        if 'text-coverletter' in request.POST:
+            if request.POST['text-coverletter']:
+                cover_letter_text = request.POST['text-coverletter']
+                user_application.cover_letter = cover_letter_text
+            else:
+                user_application.cover_letter = ''
+
+            user_application.cover_letter_file = None
+
+        elif 'add-coverletter' in request.FILES:
+            cover_letter_file = request.FILES['add-coverletter']
+            user_application.cover_letter = ''
+            user_application.cover_letter_file = cover_letter_file
+
+        if 'business_line' in request.POST or 'text-coverletter' in request.POST or 'add-coverletter' in request.FILES:
+            user_application.save()
+
+        if request.POST['next'] == 'back':
```

Comment:
Normally next is used as a param where to redirect next, so sending `back` in next and redirecting it to application_experience is confusing
- Either we should send `application_experience` instead of back in `next param`
-   OR we should rename `next` param to something more intent revealing, like `button_clicked` or anything which shows that the back button is clicked